docs/usage-guide/topics/ch11-resumption.md

```diff
@@ -25,6 +25,8 @@ Servers should set the three caching callback functions: `s2n_config_set_cache_s
 
 Clients should call `s2n_connection_get_session()` to retrieve some serialized state about the session. Then `s2n_connection_set_session()` should be called with that saved state when attempting to resume a new connection.
 
+The `cache_delete_callback()` is called when a connection encounters a fatal error, including an unexpected end-of-stream. The behavior of this callback depends on the application's implementation. If the callback is defined to delete the session cache, the application must perform a graceful TLS shutdown to enable session resumption with session caching. This is because any errors during the connection will result in that specific session being removed from the cache, making it unavailable for resumption. For more information on how to close connections, see [Closing the Connection](./ch07-io.md#closing-the-connection)
```

Comment:
You added the more accurate information, but you then repeat the less accurate version.

How about this:
```suggestion
The `cache_delete_callback` is called when a connection encounters a fatal error. This allows a server to delete a potentially corrupted or faulty session from its cache. Because an unexpected end-of-stream is considered a fatal error, an application should ensure that it performs a graceful TLS shutdown when using session caching. For more information on how to close connections, see [Closing the Connection](./ch07-io.md#closing-the-connection).
```